src/miner.cpp

```diff
@@ -116,7 +116,6 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
     pblocktemplate->vTxSigOpsCost.push_back(-1); // updated at end
 
     LOCK2(cs_main, m_mempool.cs);
-    assert(std::addressof(*::ChainActive().Tip()) == std::addressof(*m_chainstate.m_chain.Tip()));
```

Comment:
23b5ebd02e584c7bb982ccbc28bbabd927b9c57e  why would `CreateNewBlock` ever be called on the background chainstate?

Reply:
Gets called during unittests.

Reply:
But why?

Reply:
Because (unlike in actual snapshot usage) we don't have pre-prepared snapshots to use from within unittests, we have to use the same chainstate objects that we ultimately test on to generate snapshots for use during the test. As a result, the background validation chainstate is already at the snapshot base block, so in order to test appending to the bg chainstate, we have to call `CreateNewBlock` with it.

While I agree this usage isn't realistic, we don't have great alternatives within tests. I guess another option would involve using a temporary chainstate to generate the snapshot and then loading it into a new chainstatemanager object, but that sounds even hairier than what we're doing here, and much more code.